exp/runtime/hooks/api/v1alpha1/aggregate.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// AggregatableResponse represents a response object that can be constructed using individual
+// ExtensionHandlerResponses.
+// +kubebuilder:object:generate=false
+type AggregatableResponse interface {
+	runtime.Object
+
+	Aggregate(items []*ExtensionHandlerResponse) error
+}
+
+// ExtensionHandlerResponse represents ths response from a single RuntimeExtensionHandler.
+// +kubebuilder:object:generate=false
+type ExtensionHandlerResponse struct {
+	Name     string
+	Response runtime.Object
+}
+
+// aggregateExtensionResponses combines the individual ExtensionHandlerResponses into a single hook response
+// object. This function currently assumes that all the ExtensionHandlerResponses are success responses.
+func aggregateExtensionResponses(response runtime.Object, responses []*ExtensionHandlerResponse) error {
```

Comment:
Let's see what the result of https://github.com/kubernetes-sigs/cluster-api/pull/6499#discussion_r872369772 is.

If we decide to switch to a single aggregate func, let's please move the aggregation code out of this PR into a parallel PR where we migrate from the AggregatableResponse added in #6499 to a single aggregation func.

This PR here should then only contain the API types themselves